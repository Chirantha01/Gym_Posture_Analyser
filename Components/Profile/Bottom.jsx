import { StyleSheet, Text, View , Dimensions } from "react-native";
import React , { useState } from "react";
import Colors from "../Constants/Colors";
import Sizes from "../Constants/Sizes";
import Card from "../Constants/Card";
import { Entypo } from "@expo/vector-icons";
import { FontAwesome } from "@expo/vector-icons";
import Icons from 'react-native-vector-icons/MaterialCommunityIcons';
import CircularProgress from 'react-native-circular-progress-indicator';

const WIDTH =  Dimensions.get("window").width;

export default function Bottom() {
  const [value, setValue] = useState(85);

  return (
    //<ScrollView contentContainerStyle={{ alignItems: 'center', backgroundColor: "white" }}>
    <View style={styles.bottomContainer}>
      <View style={styles.completeContainer}>
        <Card
          icon={
            <Icons
              name="heart-pulse"
              size={50}
              color="red"
            />
            
          }
          cardTextOne="2"
          cardText="Health"
          style={{ backgroundColor: "#93E9BE" ,marginLeft:WIDTH*0.035 , marginRight:WIDTH*0.01}}
        />
        <Card
          icon={
            <Icons name="trophy" size={50} color="gold" />
          }
          cardTextOne="1"
          cardText="Achievements"
          style={{ backgroundColor: "#FFE4C4" , marginLeft:WIDTH*0.01 , marginRight:WIDTH*0.02}}
        />
      </View>
      <View className="flex-row m-3 rounded-3xl p-3 bg-fuchsia-200">
        <CircularProgress
          radius={60}
          value={value}
          textColor='#222'
          fontSize={20}
          valueSuffix={'%'}
          activeStrokeColor={'tomato'}
          inActiveStrokeOpacity={0.2}
          duration={4000}
        />
        <View className="justify-center mx-5">
          <Text className="font-bold text-2xl mb-2">Weakly Progress</Text>
          <Text className="font-semibold text-xl">Time Spent : 15 hours</Text>
          <Text className="font-semibold text-xl">Challenges : 12/15</Text>
        </View>
      </View>
      <View className="mt-2 flex-row">
        <Card
          icon={
            <Icons
              name="heart-pulse"
              size={50}
              color="red"
            />
            
          }
          cardTextOne="2"
          cardText="Health"
          style={{ backgroundColor: "#93E9BE" ,marginLeft:WIDTH*0.035 , marginRight:WIDTH*0.01}}
        />
        <Card
          icon={
            <Icons name="trophy" size={50} color="gold" />
          }
          cardTextOne="1"
          cardText="Achievements"
          style={{ backgroundColor: "#FFE4C4" , marginLeft:WIDTH*0.01 , marginRight:WIDTH*0.02}}
        />
      </View>
      <View style={styles.completeContainer}>
        <Card
          icon={
            <Icons
              name="heart-pulse"
              size={50}
              color="red"
            />
            
          }
          cardTextOne="2"
          cardText="Health"
          style={{ backgroundColor: "#93E9BE" ,marginLeft:WIDTH*0.035 , marginRight:WIDTH*0.01}}
        />
        <Card
          icon={
            <Icons name="trophy" size={50} color="gold" />
          }
          cardTextOne="1"
          cardText="Achievements"
          style={{ backgroundColor: "#FFE4C4" , marginLeft:WIDTH*0.01 , marginRight:WIDTH*0.02}}
        />
      </View>
      <View style={styles.bottomSection} className="p-3 bg-green-300 rounded-3xl w-8/12 mx-auto">
        <Text style={styles.bottomSectionText}>Start Exercising Now</Text>
      </View>
    </View>
    //</ScrollView>
  );
}

const styles = StyleSheet.create({
  bottomContainer: {
    marginTop: Sizes.medium,
    width:WIDTH,
    flexDirection:"column"
  },
  completeContainer: {
    flexDirection: "row",
    marginTop: Sizes.xs,
  },
  card: {
    backgroundColor: Colors.secondary,
  },
  bottomSection: {
    justifyContent: "center",
    alignItems: "center",
    marginTop: Sizes.medium,
  },
  bottomSectionText: {
    fontWeight: "bold",
    fontSize: Sizes.smedium,
    color: Colors.darkGray,
    marginBottom: 5,
    borderBottomColor: Colors.darkGray,
  }
});