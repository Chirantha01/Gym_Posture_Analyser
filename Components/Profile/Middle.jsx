import { StyleSheet, Text, View, Image } from "react-native";
import React from "react";
import { AntDesign } from "@expo/vector-icons";
import Colors from "../Constants/Colors";
import { Entypo } from "@expo/vector-icons";
import { FontAwesome } from "@expo/vector-icons";

export default function Middle() {
  return (
    <View style={styles.main}>
      <View style={styles.imageContainer}>
        <Image style={styles.image} source={require("../../assets/user.jpg")} />
        <Text style={{ fontSize: 30, color: "white", fontWeight: "bold" }}>
          Sansa Stark
        </Text>
        <Text
          style={{ fontSize: 16, fontWeight: "500" }} className="text-gray-300"
        >
          sansa2001@gmail.com
        </Text>
      </View>

      <View style={styles.middleSectionTextContainer}>
        <View style={styles.middleSectionText}>
          <Text style={styles.toptext} >Age</Text>
          <Text style={styles.bottomtext} className="text-gray-300">23</Text>
        </View>
        <View style={styles.middleSectionText}>
          <Text style={styles.toptext}>Weight</Text>
          <Text style={styles.bottomtext} className="text-gray-300">55Kg</Text>
        </View>
        <View style={styles.middleSectionText}>
          <Text style={styles.toptext}>Height</Text>
          <Text style={styles.bottomtext} className="text-gray-300">5' 6"</Text>
        </View>
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  main: {
    marginTop: 30,
  },
  imageContainer: {
    justifyContent: "center",
    alignItems: "center",
  },
  image: {
    width: 150,
    height: 150,
    borderRadius: 50,
    marginBottom: 5,
  },
  middleSectionTextContainer: {
    flexDirection: "row",
    justifyContent: "space-evenly",
    marginTop: 20,
  },
  middleSectionText: {
    justifyContent: "center",
    alignItems: "center",
  },
  toptext: {
    fontSize: 18,
    color:  Colors.white,
    fontWeight: "bold",
  },
  bottomtext: {
    fontSize: 16,
    fontWeight: "700",
  },
});
