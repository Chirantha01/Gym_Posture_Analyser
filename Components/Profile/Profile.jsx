import { StyleSheet, Text, View, Image, ScrollView, Pressable, Animated , Alert} from "react-native";
import React, { useRef } from "react";
import ImagePlaceholder from '../../assets/image-placeholder.jpg';

export default function Profile({onLogOut}) {
  const name = "John Doe";
  const email = "johnDoe@email.com";
  const birthDay = "01/01/2001";
  const age = 20;
  const height = 180;
  const weight = 70;


  // Refs for each setting animation
  const editProfileScale = useRef(new Animated.Value(1)).current;
  const changePasswordScale = useRef(new Animated.Value(1)).current;
  const logoutScale = useRef(new Animated.Value(1)).current;

  // Function to handle animations
  const animatePressIn = (scaleValue) => {
    Animated.spring(scaleValue, {
      toValue: 0.95, // Shrink to 95% of original size
      useNativeDriver: true,
    }).start();
  };

  const animatePressOut = (scaleValue) => {
    Animated.spring(scaleValue, {
      toValue: 1, // Return to original size
      friction: 3, // Control the bounce effect
      useNativeDriver: true,
    }).start();
  };

  
  const handleLogoutPress = () => {
    Alert.alert(
      "Confirm Logout",
      "Are you sure you want to log out?",
      [
        {
          text: "Cancel",
          onPress: () => console.log("Logout Cancelled"),
          style: "cancel",
        },
        {
          text: "Yes, Log Out",
          onPress: () => {onLogOut()}, // This will trigger your logout logic
        },
      ],
      { cancelable: true }
    );
  };

  return (
    <ScrollView style={styles.container}>
      <View style={styles.topContainer}>
        <View style={styles.header}>
          <Text style={styles.headerTitle}>My Profile</Text>
        </View>
        <View style={styles.imageContainer}>
          <Image source={ImagePlaceholder} style={styles.profileImage} />
        </View>
        <View>
          <Text style={styles.name}>{name}</Text>
          <Text style={styles.email}>{email}</Text>
          <Text style={styles.birthdayTitle}>
            Birthday: <Text style={styles.birthdayText}>{birthDay}</Text>
          </Text>
        </View>
      </View>
      <View style={styles.infoContainer}>
        <View style={styles.infoBlock}>
          <Text style={styles.infoLabel}>Weight</Text>
          <Text style={styles.infoText}>{weight} kg</Text>
        </View>

        {/* Vertical line */}
        <View style={styles.verticalLine} />
        
        <View style={styles.infoBlock}>
          <Text style={styles.infoLabel}>Age</Text>
          <Text style={styles.infoText}>{age}</Text>
        </View>
        
        {/* Vertical line */}
        <View style={styles.verticalLine} />
        
        <View style={styles.infoBlock}>
          <Text style={styles.infoLabel}>Height</Text>
          <Text style={styles.infoText}>{height} cm</Text>
        </View>
      </View>

      <View style={styles.bottomContainer}>
        {/* Pressable Setting Options with animation */}
        <AnimatedPressable
          scaleValue={editProfileScale}
          onPress={() => console.log("Edit Profile Pressed")}
          iconSource={require("../../assets/profileIcons/Component_39.png")}
          text="Edit my Profile"
        />

        <AnimatedPressable
          scaleValue={changePasswordScale}
          onPress={() => console.log("Change Password Pressed")}
          iconSource={require("../../assets/profileIcons/Component_41.png")}
          text="Change my Password"
        />

        <AnimatedPressable
          scaleValue={logoutScale}
          onPress={handleLogoutPress}
          iconSource={require("../../assets/profileIcons/Component_44.png")}
          text="Logout"
        />
      </View>
    </ScrollView>
  );
}

// Custom component for Animated Pressable
const AnimatedPressable = ({ scaleValue, onPress, iconSource, text }) => {
  return (
    <Animated.View style={{ transform: [{ scale: scaleValue }] }}>
      <Pressable
        onPressIn={() => animatePressIn(scaleValue)}
        onPressOut={() => animatePressOut(scaleValue)}
        onPress={onPress}
        style={styles.settings}
      >
        <Image source={iconSource} style={styles.profileIcon} />
        <Text style={styles.settingText}>{text}</Text>
      </Pressable>
    </Animated.View>
  );
};

// Functions for animation (Move these outside of the Profile function)
const animatePressIn = (scaleValue) => {
  Animated.spring(scaleValue, {
    toValue: 0.95,
    useNativeDriver: true,
  }).start();
};

const animatePressOut = (scaleValue) => {
  Animated.spring(scaleValue, {
    toValue: 1,
    friction: 3,
    useNativeDriver: true,
  }).start();
};

const styles = StyleSheet.create({
  container: {
    backgroundColor: "#232323",
  },
  topContainer: {
    backgroundColor: "#B3A0ff",
    height: 320,
    flex: 1,
  },
  header: {
    width: "100%",
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "center",
    paddingHorizontal: 20,
    paddingBottom: 20,
    paddingTop: 50,
    position: "relative",
  },
  headerTitle: {
    fontSize: 25,
    fontWeight: "bold",
    textAlign: "left",
    color: "white",
    flex: 1,
  },
  imageContainer: {
    justifyContent: "center",
    alignItems: "center",
    marginTop: -20,
  },
  profileImage: {
    width: 125,
    height: 125,
    borderRadius: 100,
  },
  name: {
    fontSize: 25,
    fontWeight: "bold",
    textAlign: "center",
    color: "white",
  },
  email: {
    fontSize: 15,
    textAlign: "center",
    color: "white",
  },
  birthdayTitle: {
    fontSize: 15,
    textAlign: "center",
    color: "white",
    fontWeight: "bold",
  },
  birthdayText: {
    fontSize: 15,
    textAlign: "center",
    color: "white",
    fontWeight: "normal",
  },
  infoContainer: {
    marginTop: -30,
    backgroundColor: "#896CFE",
    height: 60,
    width: "60%",
    flexDirection: "row",
    justifyContent: "center",
    alignItems: "center",
    alignSelf: "center",
    paddingHorizontal: 10,
    borderRadius: 10,
  },
  infoBlock: {
    flexDirection: "column",  // Stack label and value vertically
    alignItems: "center",  // Center the label and value horizontally
    paddingHorizontal: 20,
  },
  infoLabel: {
    color: "white",
    fontSize: 14,
    fontWeight: "bold",
  },
  infoText: {
    color: "white",
    fontSize: 12,
    marginTop: 4, // Space between the label and number
  },
  verticalLine: {
    height: "70%",  // Adjust the height to suit your design
    width: 1,
    backgroundColor: "white",  // Line color
  },
  bottomContainer: {
    marginTop: 20,
  },
  settings: {
    flexDirection: "row",
    paddingLeft: 20,
    paddingTop: 20,
    alignItems: "center",
  },
  profileIcon: {
    width: 30,
    height: 30,
  },
  settingText: {
    paddingTop: 7,
    fontSize: 18,
    fontWeight: "400",
    color: "white",
    paddingLeft: 20,
  },
});
