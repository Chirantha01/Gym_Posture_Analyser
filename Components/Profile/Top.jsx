import { StyleSheet, Text, View, TouchableOpacity } from "react-native";
import React from "react";
import { AntDesign } from "@expo/vector-icons";
import Colors from "../Constants/Colors";


export default function Top() {
  return (
    <View style={styles.icons}>
      <TouchableOpacity style={styles.back}>
        <AntDesign name="arrowleft" size={24} color="black" />
      </TouchableOpacity>
      <TouchableOpacity style={styles.setting}>
        <AntDesign name="setting" size={35} color="black" />
      </TouchableOpacity>
    </View>
  );
}

const styles = StyleSheet.create({
  icons: {
    flexDirection: "row",
    justifyContent: "space-between",
    alignItems: "center",
  },
  back:{
    backgroundColor: Colors.alt,
    width: 45,
    height: 45,
    borderTopLeftRadius: 10,
    borderBottomRightRadius: 10,
    justifyContent: "center",
    alignItems: "center",
  }
  
});
