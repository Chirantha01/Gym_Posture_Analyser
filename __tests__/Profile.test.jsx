import React from 'react';
import { render } from '@testing-library/react-native';
import Profile from './../Components/Profile/Profile'; // Adjust the import path as necessary

describe('Profile Component', () => {
  it('matches the snapshot', () => {
    const { toJSON } = render(<Profile />);
    expect(toJSON()).toMatchSnapshot();
  });
});