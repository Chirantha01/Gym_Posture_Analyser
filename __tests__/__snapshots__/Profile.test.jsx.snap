// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Component matches the snapshot 1`] = `
<RCTScrollView
  style={
    {
      "backgroundColor": "#232323",
    }
  }
>
  <View>
    <View
      style={
        {
          "backgroundColor": "#B3A0ff",
          "flex": 1,
          "height": 320,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
            "paddingBottom": 20,
            "paddingHorizontal": 20,
            "paddingTop": 50,
            "position": "relative",
            "width": "100%",
          }
        }
      >
        <Text
          style={
            {
              "color": "white",
              "flex": 1,
              "fontSize": 25,
              "fontWeight": "bold",
              "textAlign": "left",
            }
          }
        >
          My Profile
        </Text>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "justifyContent": "center",
            "marginTop": -20,
          }
        }
      >
        <Image
          source={
            {
              "testUri": "../../../assets/image-placeholder.jpg",
            }
          }
          style={
            {
              "borderRadius": 100,
              "height": 125,
              "width": 125,
            }
          }
        />
      </View>
      <View>
        <Text
          style={
            {
              "color": "white",
              "fontSize": 25,
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
          John Doe
        </Text>
        <Text
          style={
            {
              "color": "white",
              "fontSize": 15,
              "textAlign": "center",
            }
          }
        >
          johnDoe@email.com
        </Text>
        <Text
          style={
            {
              "color": "white",
              "fontSize": 15,
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
          Birthday: 
          <Text
            style={
              {
                "color": "white",
                "fontSize": 15,
                "fontWeight": "normal",
                "textAlign": "center",
              }
            }
          >
            01/01/2001
          </Text>
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#896CFE",
          "borderRadius": 10,
          "flexDirection": "row",
          "height": 60,
          "justifyContent": "center",
          "marginTop": -30,
          "paddingHorizontal": 10,
          "width": "60%",
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "column",
            "paddingHorizontal": 20,
          }
        }
      >
        <Text
          style={
            {
              "color": "white",
              "fontSize": 14,
              "fontWeight": "bold",
            }
          }
        >
          Weight
        </Text>
        <Text
          style={
            {
              "color": "white",
              "fontSize": 12,
              "marginTop": 4,
            }
          }
        >
          70
           kg
        </Text>
      </View>
      <View
        style={
          {
            "backgroundColor": "white",
            "height": "70%",
            "width": 1,
          }
        }
      />
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "column",
            "paddingHorizontal": 20,
          }
        }
      >
        <Text
          style={
            {
              "color": "white",
              "fontSize": 14,
              "fontWeight": "bold",
            }
          }
        >
          Age
        </Text>
        <Text
          style={
            {
              "color": "white",
              "fontSize": 12,
              "marginTop": 4,
            }
          }
        >
          20
        </Text>
      </View>
      <View
        style={
          {
            "backgroundColor": "white",
            "height": "70%",
            "width": 1,
          }
        }
      />
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "column",
            "paddingHorizontal": 20,
          }
        }
      >
        <Text
          style={
            {
              "color": "white",
              "fontSize": 14,
              "fontWeight": "bold",
            }
          }
        >
          Height
        </Text>
        <Text
          style={
            {
              "color": "white",
              "fontSize": 12,
              "marginTop": 4,
            }
          }
        >
          180
           cm
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "marginTop": 20,
        }
      }
    >
      <View
        collapsable={false}
        style={
          {
            "transform": [
              {
                "scale": 1,
              },
            ],
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingLeft": 20,
              "paddingTop": 20,
            }
          }
        >
          <Image
            source={
              {
                "testUri": "../../../assets/profileIcons/Component_39.png",
              }
            }
            style={
              {
                "height": 30,
                "width": 30,
              }
            }
          />
          <Text
            style={
              {
                "color": "white",
                "fontSize": 18,
                "fontWeight": "400",
                "paddingLeft": 20,
                "paddingTop": 7,
              }
            }
          >
            Edit my Profile
          </Text>
        </View>
      </View>
      <View
        collapsable={false}
        style={
          {
            "transform": [
              {
                "scale": 1,
              },
            ],
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingLeft": 20,
              "paddingTop": 20,
            }
          }
        >
          <Image
            source={
              {
                "testUri": "../../../assets/profileIcons/Component_41.png",
              }
            }
            style={
              {
                "height": 30,
                "width": 30,
              }
            }
          />
          <Text
            style={
              {
                "color": "white",
                "fontSize": 18,
                "fontWeight": "400",
                "paddingLeft": 20,
                "paddingTop": 7,
              }
            }
          >
            Change my Password
          </Text>
        </View>
      </View>
      <View
        collapsable={false}
        style={
          {
            "transform": [
              {
                "scale": 1,
              },
            ],
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingLeft": 20,
              "paddingTop": 20,
            }
          }
        >
          <Image
            source={
              {
                "testUri": "../../../assets/profileIcons/Component_44.png",
              }
            }
            style={
              {
                "height": 30,
                "width": 30,
              }
            }
          />
          <Text
            style={
              {
                "color": "white",
                "fontSize": 18,
                "fontWeight": "400",
                "paddingLeft": 20,
                "paddingTop": 7,
              }
            }
          >
            Logout
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
